src/app/feed/shared/feeds.service.spec.ts

```diff
@@ -209,36 +198,72 @@ describe("Service: FeedsService", () => {
 
   });
 
-  describe("waitFor",  () => {
-    it("should call feeds.find with passed id", () => {
-      spyOn(this.feeds, "find");
-
-      this.feeds.waitFor(1);
-
-      expect(this.feeds.find).toHaveBeenCalled();
-    });
-
-    it("should call feeds.find attemps + 1 times", <any>fakeAsync((): void => {
+  describe("#waitFor",  () => {
+    beforeEach(() => {
       spyOn(this.feeds, "find").and.returnValue(null);
+      this.feed = new Feed();
+      this.feed.setAttrs({id: 1});
+    });
 
-      this.feeds.waitFor(1, 5, 10);
+    it("should return a promise that resolve with asked feed", <any>fakeAsync((): void => {
+      let response: Promise<any> = this.feeds.waitFor(1, 3, 10);
 
       tick(10);
       tick(10);
```

Comment:
I don't know why but if I run `tick(20)` once fails and run `tick(10)` twice works. So for each interval cycle is necessary run one tick.

I will add a note above this code to explain this case.